### PR TITLE
Delete guides!

### LINKED ISSUE #76: Destroy guides

### PR DESCRIPTION
**Pull request by aguspina**
Closes #76

### CODE REVIEW

**Comment by faloi** [comment]
I deserve a screenshot of this...

**Comment by aguspina** [comment]
screen shot 2016-01-12 at 10 06 54 am

File: app/src/editor/GuideController.js [hunk 1]
```diff
@@ -10,12 +10,21 @@
...
-      acePreviewService, navigationService, store) {
+      acePreviewService, navigationService, store, bridge) {
+
+    $scope.topDirections = ['left', 'up'];
+    $scope.bottomDirections = ['down', 'right'];
+    $scope.isOpen = false;
+    $scope.availableModes = ['md-fling', 'md-scale'];
+    $scope.selectedMode = 'md-fling';
+    $scope.availableDirections = ['up', 'down', 'left', 'right'];
+    $scope.selectedDirection = 'up';
```
Comment (faloi) [inline app/src/editor/GuideController.js]: All this stuff is required by the toast?

File: app/src/editor/GuideController.js [hunk 2]
```diff
@@ -91,6 +100,28 @@
...
+      $mdToast.show(toast).then(function(response) {
+        if (response === 'ok' ) {
+          return;
+        } else {
+          bridge.deleteGuide($scope.selected)
+            .then(function() {
+              navigationService.goToHome();
+            })
+            .catch(function(e){
+              var message = e.data && e.data.message ? e.data.message : 'Something went wrong!';
```
Comment (faloi) [inline app/src/editor/GuideController.js]: You can use Lodash's `_.get`, since it's null safe: `_.get(e, 'data.message') || 'Sth went wrong'`

**Comment by faloi** [comment]
I'm not sure about the menu. 

The delete operation is far less frequent than the publish, and now you are penalizing me with an extra click each time. Perhaps a delete link after the selectors could do the trick.

**Comment by aguspina** [comment]
When you said 'selectors', what do you mean?

**Comment by faloi** [comment]
The combo boxes.

**Comment by aguspina** [comment]
What if I bind ctrl+s with publish action? Shall I keep this trigger button?

**Comment by faloi** [comment]
I'd love that, but my position is still the same. @flbulgarelli what do you think?

PS: I've used a hotkeys package on Classroom, check it out.

**Comment by flbulgarelli** [comment]
I think that it is true that adding an extra click is annoying. But if we add a CTRL+S, I think we have the best of both worlds: something easy to find, but also quick for the hardcore editor.

**Comment by flbulgarelli** [comment]
So :+1:  to adding the @aguspina menu, BUT with the `CTRL+S` binding

**Comment by flbulgarelli** [comment]
We should also have a shortcuts page

**Comment by faloi** [comment]
I still don't agree, but that's how democracy works.

**Comment by aguspina** [comment]
Done!

File: app/src/editor/GuideController.js [hunk 3]
```diff
@@ -91,6 +109,28 @@
...
+          return;
+        } else {
+          bridge.deleteGuide($scope.selected)
+            .then(function() {
+              navigationService.goToHome();
+            })
+            .catch(function(e){
+              var message = _.get(e, 'data.message') || 'Something went wrong'
+              $mdToast.showSimple('Ooops... ' + message);
+              console.log(e)
```
Comment (faloi) [inline app/src/editor/GuideController.js]: Is this log necessary?

**Pull request merged**